lib/shopify_app/session/session_repository.rb

```diff
@@ -3,31 +3,56 @@ class SessionRepository
     class ConfigurationError < StandardError; end
 
     class << self
-      def storage=(storage)
-        @storage = storage
+      def shop_storage=(storage)
+        @shop_storage = storage
 
-        unless storage.nil? || self.storage.respond_to?(:store) && self.storage.respond_to?(:retrieve)
-          raise ArgumentError, "storage must respond to :store and :retrieve"
+        unless storage.nil? || self.shop_storage.respond_to?(:store) && self.shop_storage.respond_to?(:retrieve)
+          raise ArgumentError, "shop storage must respond to :store and :retrieve"
         end
       end
 
-      def retrieve(id)
-        storage.retrieve(id)
+      def user_storage=(storage)
+        @user_storage = storage
+
+        unless storage.nil? || self.user_storage.respond_to?(:store) && self.user_storage.respond_to?(:retrieve)
+          raise ArgumentError, "user storage must respond to :store and :retrieve"
+        end
+      end
+
+      def retrieve_shop_session(id)
+        shop_storage.retrieve(id)
+      end
+
+      def retrieve_user_session(id)
+        user_storage.retrieve(id)
       end
 
-      def store(session, *args)
-        storage.store(session, *args)
+      def store_shop_session(session)
+        shop_storage.store(session)
       end
 
-      def storage
-        load_storage || raise(ConfigurationError.new("ShopifySessionRepository.storage is not configured!"))
+      def store_user_session(session, user)
+        user_storage.store(session, user)
+      end
+
+      def shop_storage
+        load_shop_storage || raise(ConfigurationError.new("ShopifySessionRepository.shop_storage is not configured!"))
+      end
+
+      def user_storage
+        load_user_storage
```

Comment:
absolutely not, User Session storage is not required and a change like that would immediately break every app that uses this gem and required them all to implement a User Session Storage. As well as the fact that all of the handling in the PR checks if the user session storage is nil

Reply:
Ah I see. I was assuming that `user_storage` method would only be called whenever `User` was configured and so if that happened, it would be an error. Thanks for clarifying!